src/analysis/retail/monk/mistweaver/modules/spells/MistsOfLife.tsx

```diff
@@ -0,0 +1,225 @@
+  extraRemHits: number = 0;
+  extraVivCleaves: number = 0;
+  extraVivHealing: number = 0;
+  extraVivOverhealing: number = 0;
+  envmHealingIncrease: number = 0;
+  extraEnvBonusHealing: number = 0;
+  lastVivifyCastTarget: number = 0;
+  countedMainVivifyHit: boolean = false;
+
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.hasTalent(TALENTS_MONK.MISTS_OF_LIFE_TALENT);
+    if (!this.active) {
+      return;
+    }
+    this.envmHealingIncrease = this.selectedCombatant.hasTalent(TALENTS_MONK.MIST_WRAP_TALENT.id)
+      ? 0.4
+      : 0.3;
+    this.addEventListener(
+      Events.applybuff.by(SELECTED_PLAYER).spell(TALENTS_MONK.ENVELOPING_MIST_TALENT),
+      this.handleEnvApply,
+    );
+    this.addEventListener(
+      Events.heal.by(SELECTED_PLAYER).spell(TALENTS_MONK.ENVELOPING_MIST_TALENT),
+      this.handleEnvHeal,
+    );
+    this.addEventListener(
+      Events.cast.by(SELECTED_PLAYER).spell(SPELLS.VIVIFY),
+      this.handleVivifyCast,
+    );
+    this.addEventListener(Events.heal.by(SELECTED_PLAYER).spell(SPELLS.VIVIFY), this.handleVivify);
+    this.addEventListener(
+      Events.heal.by(SELECTED_PLAYER).spell(SPELLS.RENEWING_MIST_HEAL),
+      this.handleRemHeal,
+    );
+    this.addEventListener(
+      Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.RENEWING_MIST_HEAL),
+      this.handleRemApply,
+    );
+    this.addEventListener(Events.heal.by(SELECTED_PLAYER), this.handleHeal);
+  }
+
+  handleEnvApply(event: ApplyBuffEvent) {
+    if (isFromMistsOfLife(event)) {
+      this.numEnv += 1;
+    }
+  }
+
+  handleEnvHeal(event: HealEvent) {
+    const playerId = event.targetID;
+    if (
+      !this.hotTracker.hots[playerId] ||
+      !this.hotTracker.hots[playerId][TALENTS_MONK.ENVELOPING_MIST_TALENT.id]
+    ) {
+      return;
+    }
+
+    const hot = this.hotTracker.hots[playerId][TALENTS_MONK.ENVELOPING_MIST_TALENT.id];
+    if (this.hotTracker.fromMistsOfLife(hot)) {
+      this.extraEnvmHits += 1;
+      this.extraEnvmHealing += event.amount || 0;
+      this.extraEnvmOverHealing += event.overheal || 0;
+    }
+  }
+
+  handleHeal(event: HealEvent) {
+    const targetId = event.targetID;
+    const spellId = event.ability.guid;
+    if (
+      UNAFFECTED_SPELLS.includes(spellId) ||
+      !this.hotTracker.hots[targetId] ||
+      (!this.hotTracker.hots[targetId][TALENTS_MONK.ENVELOPING_MIST_TALENT.id] &&
+        !this.hotTracker.hots[targetId][SPELLS.ENVELOPING_MIST_TFT.id])
+    ) {
+      return;
+    }
+
+    const hot = this.hotTracker.hots[targetId][TALENTS_MONK.ENVELOPING_MIST_TALENT.id];
+    if (this.hotTracker.fromMistsOfLife(hot)) {
+      this.extraEnvBonusHealing += calculateEffectiveHealing(event, this.envmHealingIncrease);
+    }
+  }
+
+  handleRemApply(event: ApplyBuffEvent | RefreshBuffEvent) {
+    const targetId = event.targetID;
+    const spellId = event.ability.guid;
+    if (
+      UNAFFECTED_SPELLS.includes(spellId) ||
```

Comment:
Isn't the check for Unaffected_Spells here unnecessary since only the event listener calling this is for ReM applications?